lightfm/_lightfm_fast.pyx.template

```diff
@@ -857,9 +861,9 @@ def fit_warp(CSRMatrix item_features,
             while sampled < lightfm.max_sampled:
 
                 sampled = sampled + 1
-                negative_item_id = (rand_r(&random_states[{thread_num}])
-                                    % item_features.rows)
-                print("Negative item id sampled")
+                negative_item_ix = (rand_r(&random_states[{thread_num}])
+                                    % len(menu_recipe_ids))
+                negative_item_id = item_ids[negative_item_ix]
```

Comment:
Just realised this is incorrect. Item_ids contain duplicates (i.e. for nnz column entry there is a row value, as per coo format), so I need to pass an array of the unique item_ids that coincide with a menu and set intersect that with item_ids.

Reply:
The correct menu_item_id is passed down into compute_representation, where the get_row_start and get_row_end methods are called to find the range of relevant CSR indices